packages/go_router/lib/src/parser.dart

```diff
@@ -0,0 +1,126 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'configuration.dart';
+import 'information_provider.dart';
+import 'logging.dart';
+import 'match.dart';
+import 'matching.dart';
+import 'redirection.dart';
+
+/// Converts between incoming URLs and a [RouteMatchList] using [RouteMatcher].
+/// Also performs redirection using [RouteRedirector].
+class GoRouteInformationParser extends RouteInformationParser<RouteMatchList> {
+  /// Creates a [GoRouteInformationParser].
+  GoRouteInformationParser({
+    required this.configuration,
+    this.debugRequireGoRouteInformationProvider = false,
+  })  : matcher = RouteMatcher(configuration),
+        redirector = redirect;
+
+  /// The route configuration for the app.
+  final RouteConfiguration configuration;
+
+  /// The route matcher.
+  final RouteMatcher matcher;
+
+  /// The route redirector.
+  final RouteRedirector redirector;
+
+  /// A debug property to assert [GoRouteInformationProvider] is in use along
+  /// with this parser.
+  ///
+  /// An assertion error will be thrown if this property set to true and the
+  /// [GoRouteInformationProvider] is not in use.
+  ///
+  /// Defaults to false.
+  final bool debugRequireGoRouteInformationProvider;
+
+  /// Called by the [Router]. The
+  @override
+  Future<RouteMatchList> parseRouteInformation(
+    RouteInformation routeInformation,
+  ) {
+    assert(() {
+      if (debugRequireGoRouteInformationProvider) {
+        assert(
+            routeInformation is DebugGoRouteInformation,
+            'This GoRouteInformationParser needs to be used with '
+            'GoRouteInformationProvider, did you forget to pass in '
+            'GoRouter.routeInformationProvider to the Router constructor?');
```

Comment:
Does this use the same style guide as the flutter repo? If so, I think this should be like this:

```suggestion
        assert(
            routeInformation is DebugGoRouteInformation,
            'This GoRouteInformationParser needs to be used with '
            'GoRouteInformationProvider, did you forget to pass in '
            'GoRouter.routeInformationProvider to the Router constructor?',
        );
```